test/library/etcd.go

```diff
@@ -0,0 +1,149 @@
+package library
+
+import (
+	"bufio"
+	"bytes"
+	"context"
+	"encoding/hex"
+	"fmt"
+	"os/exec"
+	"strconv"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/coreos/etcd/clientv3"
+	"github.com/stretchr/testify/require"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apiserver/pkg/storage"
+	"k8s.io/client-go/kubernetes"
+	restclient "k8s.io/client-go/rest"
+)
+
+var protoEncodingPrefix = []byte{0x6b, 0x38, 0x73, 0x00}
+
+const (
+	protoEncryptedDataPrefix     = "k8s:enc:"
+	aesCBCTransformerPrefixV1    = "k8s:enc:aescbc:v1:"
+	secretboxTransformerPrefixV1 = "k8s:enc:secretbox:v1:"
+)
+
+func NewEtcdKVMust(t *testing.T, kubeClient kubernetes.Interface) (clientv3.KV, func()) {
+	t.Helper()
+	kv, done, err := NewEtcdKV(kubeClient)
+	require.NoError(t, err)
+	return kv, done
+}
+
+func NewEtcdKV(kubeClient kubernetes.Interface) (clientv3.KV, func(), error) {
+	ctx, cancel := context.WithCancel(context.Background())
+	cmd := exec.CommandContext(ctx, "oc", "port-forward", "service/etcd", ":2379", "-n", "openshift-etcd")
+
+	done := func() {
+		cancel()
+		_ = cmd.Wait() // wait to clean up resources but ignore returned error since cancel kills the process
+	}
+
+	var err error // so we can clean up on error
+	defer func() {
+		if err != nil {
+			done()
+		}
+	}()
+
+	stdOut, err := cmd.StdoutPipe()
+	if err != nil {
+		return nil, nil, err
+	}
+
+	if err = cmd.Start(); err != nil {
+		return nil, nil, err
+	}
+
+	scanner := bufio.NewScanner(stdOut)
+	if !scanner.Scan() {
+		return nil, nil, fmt.Errorf("failed to scan port forward std out")
+	}
+	if err = scanner.Err(); err != nil {
+		return nil, nil, err
+	}
+	output := scanner.Text()
+
+	port := strings.TrimSuffix(strings.TrimPrefix(output, "Forwarding from 127.0.0.1:"), " -> 2379")
+	_, err = strconv.Atoi(port)
```

Comment:
this seems to be unnecessary since you know the port, right?

Reply:
No, we are asking the command to find a free random port for us.  We then parse out the value it gives us.